### PR TITLE
Update config doc with Java client

### LINKED ISSUE #11125: Add SignalR Java Client to Configure Client Options
https://docs.microsoft.com/en-us/aspnet/core/signalr/configuration?view=aspnetcore-2.1#configure-client-options

We need to add guidance for how to configure client options for the Java client.

### PR DESCRIPTION
**Pull request by mikaelm12**
Fixes #11125 

Adds guidance on how to configure the various options on the SignalR Java client. 

[Internal review](https://review.docs.microsoft.com/en-us/aspnet/core/signalr/configuration?view=aspnetcore-2.1&branch=pr-en-us-11143)

### CODE REVIEW

File: aspnetcore/signalr/configuration.md [hunk 1]
```diff
@@ -197,6 +199,17 @@ let connection = new signalR.HubConnectionBuilder()
     .build();
 ```
 
+In the Java client the transport is selected with the `withTransport` method on the HttpHubConnectionBuilder. The Java client defaults to using the WebSockets transport.
```
Comment (BrennanConroy) [inline aspnetcore/signalr/configuration.md]: Need to moniker this for 3.0 only

File: aspnetcore/signalr/configuration.md [hunk 2]
```diff
@@ -227,34 +240,45 @@ let connection = new signalR.HubConnectionBuilder()
...
-| `ServerTimeout` | `serverTimeoutInMilliseconds` | 30 seconds (30,000 milliseconds) | Timeout for server activity. If the server hasn't sent a message in this interval, the client considers the server disconnected and triggers the `Closed` event (`onclose` in JavaScript). This value must be large enough for a ping message to be sent from the server **and** received by the client within the timeout interval. The recommended value is a number at least double the server's `KeepAliveInterval` value, to allow time for pings to arrive. |
-| `HandshakeTimeout` | Not configurable | 15 seconds | Timeout for initial server handshake. If the server doesn't send a handshake response in this interval, the client cancels the handshake and triggers the `Closed` event (`onclose` in JavaScript). This is an advanced setting that should only be modified if handshake timeout errors are occurring due to severe network latency. For more detail on the Handshake process, see the [SignalR Hub Protocol Specification](https://github.com/aspnet/SignalR/blob/master/specs/HubProtocol.md). |
+| .NET Option | JavaScript Option | Java Option | Default Value | Description |
```
Comment (BrennanConroy) [inline aspnetcore/signalr/configuration.md]: We'll probably want to refactor this table now that it's getting bigger.

File: aspnetcore/signalr/configuration.md [hunk 3]
```diff
@@ -281,6 +305,17 @@ let connection = new signalR.HubConnectionBuilder()
     .build();
 ```
 
+In the Java client, these options can be configured with the methods on the `HttpHubConnectionBuilder` returned from the `HubConnectionBuilder.create("HUB URL")`
+
+
+```java
+    HubConnection hubConnection = HubConnectionBuilder.create("https://example.com/myhub")
+            .withHeader("Foo", "Bar")
+            .shoudldSkipNegotiate(true)
```
Comment (BrennanConroy) [inline aspnetcore/signalr/configuration.md]: shouldSkipNegotiate

File: aspnetcore/signalr/configuration.md [hunk 4]
```diff
@@ -132,7 +132,7 @@ The WebSocket transport has additional options that can be configured using the
 
 ## Configure client options
 
-Client options can be configured on the `HubConnectionBuilder` type (available in both .NET and JavaScript clients), as well as on the `HubConnection` itself.
+Client options can be configured on the `HubConnectionBuilder` type (available in the .NET and Javascipt clients. It is also available on the Java client but the `HttpHubConnectionBuilder` subclass is what contains the builder configuation options.), as well as on the `HubConnection` itself.
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]:
```suggestion
Client options can be configured on the `HubConnectionBuilder` type (available in the .NET and JavaScript clients). It's also available in the Java client, but the `HttpHubConnectionBuilder` subclass is what contains the builder configuration options, as well as on the `HubConnection` itself.
```

File: aspnetcore/signalr/configuration.md [hunk 5]
```diff
@@ -162,6 +162,8 @@ let connection = new signalR.HubConnectionBuilder()
     .build();
 ```
 
+See [Logging in SignalR](xref:signalr/java-client#Add-logging) for how to enable logging in the Java client
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]:
```suggestion
For details on how to enable logging in the Java client, see [Logging in SignalR](xref:signalr/java-client#Add-logging).
```

File: aspnetcore/signalr/configuration.md (hunk 1 above)
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]:
```suggestion
In the Java client, the transport is selected with the `withTransport` method on the `HttpHubConnectionBuilder`. The Java client defaults to using the WebSockets transport.
```

File: aspnetcore/signalr/configuration.md [hunk 6]
```diff
@@ -197,6 +199,17 @@ let connection = new signalR.HubConnectionBuilder()
     .build();
 ```
 
+In the Java client the transport is selected with the `withTransport` method on the HttpHubConnectionBuilder. The Java client defaults to using the WebSockets transport.
+
+```java
+        HubConnection hubConnection = HubConnectionBuilder.create("https://example.com/myhub")
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]: dedent this code snippet

File: aspnetcore/signalr/configuration.md [hunk 7]
```diff
@@ -281,6 +305,17 @@ let connection = new signalR.HubConnectionBuilder()
     .build();
 ```
 
+In the Java client, these options can be configured with the methods on the `HttpHubConnectionBuilder` returned from the `HubConnectionBuilder.create("HUB URL")`
+
+
+```java
+    HubConnection hubConnection = HubConnectionBuilder.create("https://example.com/myhub")
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]: dedent this code snippet

File: aspnetcore/signalr/configuration.md (hunk 2 above)
Comment (scottaddie) [reply aspnetcore/signalr/configuration.md]: I agree with Brennan. What do you think about using 3 tabs: .NET, JavaScript, and Java?

Comment (mikaelm12) [reply aspnetcore/signalr/configuration.md]: > 3 tabs: .NET, JavaScript, and Java?

Just want to clarify,  under Configuration in the TOC having separate sections for each client? Or something else?

Comment (scottaddie) [reply aspnetcore/signalr/configuration.md]: I'm imagining actual tabs (see https://docs.microsoft.com/aspnet/core/tutorials/first-web-api?view=aspnetcore-2.2&tabs=visual-studio#create-a-web-project for an example). The syntax for the tabs would be:

```markdown
[.NET](#tab/dotnet)

.NET-specific table here

[JavaScript](#tab/javascript)

JavaScript-specific table here

[Java](#tab/java)

Java-specific table here

---
```

Comment (mikaelm12) [reply aspnetcore/signalr/configuration.md]: Oh, awesome. Thanks for the clarification. 👍

Comment (bradygaster) [reply aspnetcore/signalr/configuration.md]: ver ver cool

**Comment by bradygaster** [comment]
lgtm

File: aspnetcore/signalr/configuration.md (hunk 4 above)
Comment (halter73) [reply aspnetcore/signalr/configuration.md]: :+1: I also like removing the parens entirely:

```suggestion
Client options can be configured on the `HubConnectionBuilder` type available in the .NET and JavaScript clients. It's also available in the Java client, but the `HttpHubConnectionBuilder` subclass is what contains the builder configuration options, as well as on the `HubConnection` itself.
```

File: aspnetcore/signalr/configuration.md [hunk 8]
```diff
@@ -227,34 +240,45 @@ let connection = new signalR.HubConnectionBuilder()
     .build();
 ```
 
+
+In the SignalR Java client, you can configure a bearer token to use for authentication by providing an "access token factory" to the [HttpHubConnectionBuilder](/java/api/com.microsoft.signalr._http_hub_connection_builder?view=aspnet-signalr-java). Use [withAccessTokenFactory](/java/api/com.microsoft.signalr._http_hub_connection_builder.withaccesstokenprovider?view=aspnet-signalr-java#com_microsoft_signalr__http_hub_connection_builder_withAccessTokenProvider_Single_String__) to provide an [RxJava](https://github.com/ReactiveX/RxJava) [Single<String>](http://reactivex.io/documentation/single.html). With a call to [Single.defer](http://reactivex.io/RxJava/javadoc/io/reactivex/Single.html#defer-java.util.concurrent.Callable-), you can write logic to produce access tokens for your client.
```
Comment (halter73) [inline aspnetcore/signalr/configuration.md]: Why put quotes around "access token factory"?

File: aspnetcore/signalr/configuration.md [hunk 9]
```diff
@@ -227,34 +240,45 @@ let connection = new signalR.HubConnectionBuilder()
...
+| `HandshakeTimeout` | Not configurable |  `withHandshakeResponseTimeout` | 15 seconds | Timeout for initial server handshake. If the server doesn't send a handshake response in this interval, the client cancels the handshake and triggers the `Closed` event (`onclose` in JavaScript). This is an advanced setting that should only be modified if handshake timeout errors are occurring due to severe network latency. For more detail on the Handshake process, see the [SignalR Hub Protocol Specification](https://github.com/aspnet/SignalR/blob/master/specs/HubProtocol.md). |
 
 In the .NET Client, timeout values are specified as `TimeSpan` values. In the JavaScript client, timeout values are specified as a number indicating the duration in milliseconds.
```
Comment (halter73) [inline aspnetcore/signalr/configuration.md]: What type is used f Java client timeouts?

File: aspnetcore/signalr/configuration.md [hunk 10]
```diff
@@ -227,34 +240,45 @@ let connection = new signalR.HubConnectionBuilder()
...
+| `ServerTimeout` | `serverTimeoutInMilliseconds` | `getServerTimeout` `setServerTimeout` | 30 seconds (30,000 milliseconds) | Timeout for server activity. If the server hasn't sent a message in this interval, the client considers the server disconnected and triggers the `Closed` event (`onclose` in JavaScript). This value must be large enough for a ping message to be sent from the server **and** received by the client within the timeout interval. The recommended value is a number at least double the server's `KeepAliveInterval` value, to allow time for pings to arrive. |
+| `HandshakeTimeout` | Not configurable |  `withHandshakeResponseTimeout` | 15 seconds | Timeout for initial server handshake. If the server doesn't send a handshake response in this interval, the client cancels the handshake and triggers the `Closed` event (`onclose` in JavaScript). This is an advanced setting that should only be modified if handshake timeout errors are occurring due to severe network latency. For more detail on the Handshake process, see the [SignalR Hub Protocol Specification](https://github.com/aspnet/SignalR/blob/master/specs/HubProtocol.md). |
```
Comment (halter73) [inline aspnetcore/signalr/configuration.md]: > If the server doesn't send a handshake response in this interval, the client cancels the handshake and triggers the `Closed` event (`onclose` in JavaScript)

Make sure to mention `onClosed` for Java.

File: aspnetcore/signalr/configuration.md [hunk 11]
```diff
@@ -227,34 +240,45 @@ let connection = new signalR.HubConnectionBuilder()
...
-| `UseDefaultCredentials` | Not configurable * | `false` | Set this boolean to send the default credentials for HTTP and WebSockets requests. This enables the use of Windows authentication. |
-| `WebSocketConfiguration` | Not configurable * | `null` | A delegate that can be used to configure additional WebSocket options. Receives an instance of [ClientWebSocketOptions](/dotnet/api/system.net.websockets.clientwebsocketoptions) that can be used to configure the options. |
+| `Cookies` | Not configurable * | Not configurable | Empty | A collection of HTTP cookies to send with every HTTP request. |
+| `Credentials` | Not configurable * | Not configurable | Empty | Credentials to send with every HTTP request. |
+| `CloseTimeout` | Not configurable * | Not configurable | 5 seconds | WebSockets only. The maximum amount of time the client waits after closing for the server to acknowledge the close request. If the server doesn't acknowledge the close within this time, the client disconnects. |
+| `Headers` | Not configurable * | `withHeader``withHeaders` | Empty | A dictionary of additional HTTP headers to send with every HTTP request. |
```
Comment (halter73) [inline aspnetcore/signalr/configuration.md]: I wonder if `Map` would be more familiar than dictionary to Java devs.

File: aspnetcore/signalr/configuration.md (hunk 8 above)
Comment (mikaelm12) [reply aspnetcore/signalr/configuration.md]: I copy pasted this from somewhere else. Quotes not needed

File: aspnetcore/signalr/configuration.md (hunk 9 above)
Comment (mikaelm12) [reply aspnetcore/signalr/configuration.md]: `long`s representing milliseconds. I'll add it

**Comment by mikaelm12** [comment]
Tab formatting is messed up atm.

File: aspnetcore/signalr/configuration.md [hunk 12]
```diff
@@ -227,34 +250,64 @@ let connection = new signalR.HubConnectionBuilder()
...
+| ----------- | ------------- | ----------- |
+| `ServerTimeout` | 30 seconds (30,000 milliseconds) | Timeout for server activity. If the server hasn't sent a message in this interval, the client considers the server disconnected and triggers the `Closed` event (`onclose` in JavaScript). This value must be large enough for a ping message to be sent from the server **and** received by the client within the timeout interval. The recommended value is a number at least double the server's `KeepAliveInterval` value, to allow time for pings to arrive. |
+| `HandshakeTimeout` | 15 seconds | Timeout for initial server handshake. If the server doesn't send a handshake response in this interval, the client cancels the handshake and triggers the `Closed` event (`onclose` in JavaScript). This is an advanced setting that should only be modified if handshake timeout errors are occurring due to severe network latency. For more detail on the Handshake process, see the [SignalR Hub Protocol Specification](https://github.com/aspnet/SignalR/blob/master/specs/HubProtocol.md). |
+
+In the .NET Client, timeout values are specified as `TimeSpan` values.
+
+[Javascript](#tab/javascript)
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]:
```suggestion
# [JavaScript](#tab/javascript)

```

File: aspnetcore/signalr/configuration.md [hunk 13]
```diff
@@ -227,34 +250,64 @@ let connection = new signalR.HubConnectionBuilder()
...
+| `HandshakeTimeout` | Not configurable |  `withHandshakeResponseTimeout` | 15 seconds | Timeout for initial server handshake. If the server doesn't send a handshake response in this interval, the client cancels the handshake and triggers the `Closed` event (`onclose` in JavaScript). This is an advanced setting that should only be modified if handshake timeout errors are occurring due to severe network latency. For more detail on the Handshake process, see the [SignalR Hub Protocol Specification](https://github.com/aspnet/SignalR/blob/master/specs/HubProtocol.md). |
 
 In the .NET Client, timeout values are specified as `TimeSpan` values. In the JavaScript client, timeout values are specified as a number indicating the duration in milliseconds.
 
+[.NET](#tab/dotnet)
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]:
```suggestion
# [.NET](#tab/dotnet)

```

File: aspnetcore/signalr/configuration.md [hunk 14]
```diff
@@ -227,34 +250,64 @@ let connection = new signalR.HubConnectionBuilder()
...
+
+In the .NET Client, timeout values are specified as `TimeSpan` values.
+
+[Javascript](#tab/javascript)
+| JavaScript Option | Default Value | Description |
+| ----------------- | ------------- | ----------- |
+ `serverTimeoutInMilliseconds` | 30 seconds (30,000 milliseconds) | Timeout for server activity. If the server hasn't sent a message in this interval, the client considers the server disconnected and triggers the `onclose` event. This value must be large enough for a ping message to be sent from the server **and** received by the client within the timeout interval. The recommended value is a number at least double the server's `KeepAliveInterval` value, to allow time for pings to arrive. |
+
+ [Java](#tab/java)
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]:
```suggestion
# [Java](#tab/java)

```

File: aspnetcore/signalr/configuration.md [hunk 15]
```diff
@@ -227,34 +250,64 @@ let connection = new signalR.HubConnectionBuilder()
...
+| Java Option | Default Value | Description |
+| ----------- | ------------- | ----------- |
+|`getServerTimeout` `setServerTimeout` | 30 seconds (30,000 milliseconds) | Timeout for server activity. If the server hasn't sent a message in this interval, the client considers the server disconnected and triggers the `onClose` event. This value must be large enough for a ping message to be sent from the server **and** received by the client within the timeout interval. The recommended value is a number at least double the server's `KeepAliveInterval` value, to allow time for pings to arrive. |
+| `withHandshakeResponseTimeout` | 15 seconds | Timeout for initial server handshake. If the server doesn't send a handshake response in this interval, the client cancels the handshake and triggers the `onClose` event. This is an advanced setting that should only be modified if handshake timeout errors are occurring due to severe network latency. For more detail on the Handshake process, see the [SignalR Hub Protocol Specification](https://github.com/aspnet/SignalR/blob/master/specs/HubProtocol.md). |
+
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]: Terminate the tab set with 3 dashes.

```suggestion

---

```

File: aspnetcore/signalr/configuration.md [hunk 16]
```diff
@@ -227,34 +250,64 @@ let connection = new signalR.HubConnectionBuilder()
...
+In the .NET Client, timeout values are specified as `TimeSpan` values.
+
+[Javascript](#tab/javascript)
+| JavaScript Option | Default Value | Description |
+| ----------------- | ------------- | ----------- |
+ `serverTimeoutInMilliseconds` | 30 seconds (30,000 milliseconds) | Timeout for server activity. If the server hasn't sent a message in this interval, the client considers the server disconnected and triggers the `onclose` event. This value must be large enough for a ping message to be sent from the server **and** received by the client within the timeout interval. The recommended value is a number at least double the server's `KeepAliveInterval` value, to allow time for pings to arrive. |
+
+ [Java](#tab/java)
+| Java Option | Default Value | Description |
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]:
```suggestion
| Option | Default value | Description |
```

File: aspnetcore/signalr/configuration.md [hunk 17]
```diff
@@ -227,34 +250,64 @@ let connection = new signalR.HubConnectionBuilder()
...
+| `ServerTimeout` | 30 seconds (30,000 milliseconds) | Timeout for server activity. If the server hasn't sent a message in this interval, the client considers the server disconnected and triggers the `Closed` event (`onclose` in JavaScript). This value must be large enough for a ping message to be sent from the server **and** received by the client within the timeout interval. The recommended value is a number at least double the server's `KeepAliveInterval` value, to allow time for pings to arrive. |
+| `HandshakeTimeout` | 15 seconds | Timeout for initial server handshake. If the server doesn't send a handshake response in this interval, the client cancels the handshake and triggers the `Closed` event (`onclose` in JavaScript). This is an advanced setting that should only be modified if handshake timeout errors are occurring due to severe network latency. For more detail on the Handshake process, see the [SignalR Hub Protocol Specification](https://github.com/aspnet/SignalR/blob/master/specs/HubProtocol.md). |
+
+In the .NET Client, timeout values are specified as `TimeSpan` values.
+
+[Javascript](#tab/javascript)
+| JavaScript Option | Default Value | Description |
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]:
```suggestion
| Option | Default value | Description |
```

File: aspnetcore/signalr/configuration.md [hunk 18]
```diff
@@ -227,34 +250,64 @@ let connection = new signalR.HubConnectionBuilder()
...
+| `HandshakeTimeout` | Not configurable |  `withHandshakeResponseTimeout` | 15 seconds | Timeout for initial server handshake. If the server doesn't send a handshake response in this interval, the client cancels the handshake and triggers the `Closed` event (`onclose` in JavaScript). This is an advanced setting that should only be modified if handshake timeout errors are occurring due to severe network latency. For more detail on the Handshake process, see the [SignalR Hub Protocol Specification](https://github.com/aspnet/SignalR/blob/master/specs/HubProtocol.md). |
 
 In the .NET Client, timeout values are specified as `TimeSpan` values. In the JavaScript client, timeout values are specified as a number indicating the duration in milliseconds.
 
+[.NET](#tab/dotnet)
+| .NET Option | Default Value | Description |
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]:
```suggestion
| Option | Default value | Description |
```

File: aspnetcore/signalr/configuration.md (hunk 2 above)
Comment (BrennanConroy) [reply aspnetcore/signalr/configuration.md]: Remove this section now

File: aspnetcore/signalr/configuration.md [hunk 19]
```diff
@@ -227,34 +250,69 @@ let connection = new signalR.HubConnectionBuilder()
...
 ### Configure additional options
 
-Additional options can be configured in the `WithUrl` (`withUrl` in JavaScript) method on `HubConnectionBuilder`:
+Additional options can be configured in the `WithUrl` (`withUrl` in JavaScript) method on `HubConnectionBuilder` or on the various configuration APIs on the `HttpHubConnectionBuilder` in the Java client:
 
-| .NET Option | JavaScript Option | Default Value | Description |
-| ----------- | ----------------- | ------------- | ----------- |
-| `AccessTokenProvider` | `accessTokenFactory` | `null` | A function returning a string that is provided as a Bearer authentication token in HTTP requests. |
-| `SkipNegotiation` | `skipNegotiation` | `false` | Set this to `true` to skip the negotiation step. **Only supported when the WebSockets transport is the only enabled transport**. This setting can't be enabled when using the Azure SignalR Service. |
+| .NET Option | JavaScript Option | Java Option | Default Value | Description |
```
Comment (BrennanConroy) [inline aspnetcore/signalr/configuration.md]: Tabs

File: aspnetcore/signalr/configuration.md [hunk 20]
```diff
@@ -197,6 +199,27 @@ let connection = new signalR.HubConnectionBuilder()
...
+```java
+HubConnection hubConnection = HubConnectionBuilder.create("https://example.com/myhub")
+    .withTransport(TransportEnum.WEBSOCKETS)
+    .build();
+```
+
+::: moniker-end
+
+> [!NOTE]
+> The SignalR Java client doesn't support transport fallback yet.
```
Comment (BrennanConroy) [inline aspnetcore/signalr/configuration.md]: Move this into the 3.0 range

File: aspnetcore/signalr/configuration.md [hunk 21]
```diff
@@ -227,34 +250,69 @@ let connection = new signalR.HubConnectionBuilder()
...
-| `UseDefaultCredentials` | Not configurable * | `false` | Set this boolean to send the default credentials for HTTP and WebSockets requests. This enables the use of Windows authentication. |
-| `WebSocketConfiguration` | Not configurable * | `null` | A delegate that can be used to configure additional WebSocket options. Receives an instance of [ClientWebSocketOptions](/dotnet/api/system.net.websockets.clientwebsocketoptions) that can be used to configure the options. |
+| `Cookies` | Not configurable * | Not configurable | Empty | A collection of HTTP cookies to send with every HTTP request. |
+| `Credentials` | Not configurable * | Not configurable | Empty | Credentials to send with every HTTP request. |
+| `CloseTimeout` | Not configurable * | Not configurable | 5 seconds | WebSockets only. The maximum amount of time the client waits after closing for the server to acknowledge the close request. If the server doesn't acknowledge the close within this time, the client disconnects. |
+| `Headers` | Not configurable * | `withHeader``withHeaders` | Empty | A Map of additional HTTP headers to send with every HTTP request. |
```
Comment (BrennanConroy) [inline aspnetcore/signalr/configuration.md]: `withHeader``withHeaders` needs a space

File: aspnetcore/signalr/configuration.md [hunk 22]
```diff
@@ -227,34 +250,69 @@ let connection = new signalR.HubConnectionBuilder()
...
+
+In the .NET Client, timeout values are specified as `TimeSpan` values.
+
+# [JavaScript](#tab/javascript)
+
+| Option | Default value | Description |
+| ----------------- | ------------- | ----------- |
+ `serverTimeoutInMilliseconds` | 30 seconds (30,000 milliseconds) | Timeout for server activity. If the server hasn't sent a message in this interval, the client considers the server disconnected and triggers the `onclose` event. This value must be large enough for a ping message to be sent from the server **and** received by the client within the timeout interval. The recommended value is a number at least double the server's `KeepAliveInterval` value, to allow time for pings to arrive. |
```
Comment (scottaddie) [inline aspnetcore/signalr/configuration.md]:
```suggestion
| `serverTimeoutInMilliseconds` | 30 seconds (30,000 milliseconds) | Timeout for server activity. If the server hasn't sent a message in this interval, the client considers the server disconnected and triggers the `onclose` event. This value must be large enough for a ping message to be sent from the server **and** received by the client within the timeout interval. The recommended value is a number at least double the server's `KeepAliveInterval` value, to allow time for pings to arrive. |

In the JavaScript client, timeout values are specified as a number indicating the duration in milliseconds.
```

File: aspnetcore/signalr/configuration.md (hunk 2 above)
Comment (scottaddie) [reply aspnetcore/signalr/configuration.md]: I agree with Brennan's suggestion ☝️ . Also remove the 2 sentences below this table.

File: aspnetcore/signalr/configuration.md [hunk 23]
```diff
@@ -177,6 +177,8 @@ Log levels available to the JavaScript client are listed below. Setting the log
 | `Debug` | Diagnostic messages useful for debugging. |
 | `Trace` | Very detailed diagnostic messages designed for diagnosing specific issues. |
 
+For details on how to enable logging in the Java client, see [Logging in SignalR](xref:signalr/java-client#Add-logging).
```
Comment (BrennanConroy) [inline aspnetcore/signalr/configuration.md]: Probably not in scope for this PR, but this sentence is kind of hidden, we might want to rework the logging stuff at some point

Comment (mikaelm12) [reply aspnetcore/signalr/configuration.md]: I'm just gonna copy over the logging instructions from the other page. Better to just have the info here to match the other sections.

**Pull request merged**